### PR TITLE
Consolidate usage of pathlib.Path instead of os.path

### LINKED ISSUE #196: Consistent use of pathlib vs os.path
Consider to switch all paths to pathlib.Path, currently a mix between pathlib.Path and os.path

### PR DESCRIPTION
**Pull request by vantaka2**
Closes #196 

Bare with me on this one, first time using pathlib and still trying to get a full understanding of the repository 

Would appreciate some feedback from one of the maintainers to see I'm on the right track with the changes so far!

### CODE REVIEW

File: webviz_config/_build_webviz.py [hunk 1]
```diff
@@ -22,22 +24,22 @@ def build_webviz(args: argparse.Namespace) -> None:
         raise ValueError(f"Theme `{args.theme}` is not installed.")
 
     if args.portable is None:
-        build_directory = tempfile.mkdtemp()
+        build_directory = pathlib.Path(tempfile.mkdtemp()).resolve().parent
```
Comment (anders-kiaer) [inline webviz_config/_build_webviz.py:27]: `tempfile.mkdtemp()` already returns an absolute path, so don't need to call `resolve()`. Also you want the folder itself, and not the parent folder.

```suggestion
        build_directory = pathlib.Path(tempfile.mkdtemp())
```

File: webviz_config/_build_webviz.py [hunk 2]
```diff
@@ -22,22 +24,22 @@ def build_webviz(args: argparse.Namespace) -> None:
...
     else:
-        build_directory = args.portable
+        build_directory = pathlib.Path(args.portable).resolve().parent
```
Comment (anders-kiaer) [inline webviz_config/_build_webviz.py:29]:
```suggestion
        build_directory = pathlib.Path(args.portable).resolve()
```

File: webviz_config/_build_webviz.py [hunk 3]
```diff
@@ -22,22 +24,22 @@ def build_webviz(args: argparse.Namespace) -> None:
...
-        shutil.copy(os.path.join(STATIC_FOLDER, filename), build_directory)
+        shutil.copy(STATIC_FOLDER / filename, build_directory)
 
     for asset in installed_themes[args.theme].assets:
-        shutil.copy(asset, os.path.join(build_directory, "assets"))
+        shutil.copy(asset, build_directory / "assets")
 
-    with open(os.path.join(build_directory, "theme_settings.json"), "w") as filehandle:
+    with open(build_directory / "theme_settings.json", "w") as filehandle:
         filehandle.write(installed_themes[args.theme].to_json())
```
Comment (anders-kiaer) [inline webviz_config/_build_webviz.py:43]: Here we could perhaps utilize the [`path.write_text()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text) method.

```suggestion
    (build_directory / "theme_settings.json").write_text(installed_themes[args.theme].to_json())
```

File: webviz_config/_build_webviz.py [hunk 4]
```diff
@@ -64,7 +66,7 @@ def build_webviz(args: argparse.Namespace) -> None:
                     "error message and traceback above"
                 )
 
-            os.remove(os.path.join(build_directory, "copy_data.py"))
+            os.remove(build_directory / "copy_data.py")
```
Comment (anders-kiaer) [inline webviz_config/_build_webviz.py:69]: Here we could perhaps use [`unlink`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink).

```suggestion
            (build_directory / "copy_data.py").unlink()
```

File: webviz_config/_write_script.py [hunk 5]
```diff
@@ -36,7 +36,7 @@ def write_script(
 
     template = template_environment.get_template(template_filename)
 
-    with open(os.path.join(build_directory, output_filename), "w") as filehandle:
+    with open(build_directory / output_filename), "w") as filehandle:
         filehandle.write(template.render(configuration))
```
Comment (anders-kiaer) [inline webviz_config/_write_script.py:40]:
```suggestion
    (build_directory / output_filename).write_text(template.render(configuration))
```

**Review by anders-kiaer — DISMISSED** [review]
Thanks @vantaka2 for the contribution. Looks good so far 👍 Added some suggestions. In addition, simply run `black webviz_config` in order to get 🟢 from CI.

**Comment by anders-kiaer** [comment]
Thanks again for opening this PR @vantaka2. I will try to to get this to the 🥅 the next days in order to solve #196. 🙂

**Review by HansKallekleiv — APPROVED** [review]
lgtm 👍

**Pull request merged**